tools/docker/run.sh

```diff
@@ -64,7 +65,8 @@ main() {
             -n | --name)        NAME="$2"; shift; shift ;;
             -u | --unique-id)   UNIQUE_ID="$2"; shift; shift ;;
             -t | --tag)         TAG=":$2"; shift; shift ;;
-            -p | --port)        PORT=":$2"; shift; shift ;;
+            -p | --port)        PORT="$2"; shift; shift ;;
+            -P | --publish)     PUBLISH="-P"; shift ;;
```

Comment:
Publish just makes these ports accessible to the host.
I do prefer random ports (which we can later read by `docker port`) instead of managing them myself.

Reply:
`--publish` (which is not the option you used) will make it accept connections from whatever IP you specified.
`--publish-all` will make it accept connections from anywhere.

Still from `man docker-run`:
>When set to true publish all exposed ports to the host interfaces. The default is false. If the operator uses -P (or -p) then Docker will make the exposed port accessible on the host and
       **the ports will be available to any client that can reach the host**. When using -P, Docker will bind any exposed port to a random port on the host within an ephemeral port range defined by
>       /proc/sys/net/ipv4/ip_local_port_range. To find the mapping between the host ports and the exposed ports, use docker port(1).

In other words, you _have_ to use `--publish 127.0.0.1::<capi_port>` unless you want the CAPI listener to be available to anyone on your local network.

(you will still have random ports on the host if you specify the IP and the container port only)

Reply:
Updated the script to allow publishing ports on the localhost only.